primeqa/mrc/run_mrc.py

```diff
@@ -315,162 +328,191 @@ def main():
                 f"Checkpoint detected, resuming training at {last_checkpoint}. To avoid this behavior, change "
                 "the `--output_dir` or add `--overwrite_output_dir` to train from scratch."
             )
-
-    task_heads = task_args.task_heads
-    config = AutoConfig.from_pretrained(
-        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
-        cache_dir=model_args.cache_dir,
-    )
-    tokenizer = AutoTokenizer.from_pretrained(
-        model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
-        cache_dir=model_args.cache_dir,
-        use_fast=True,
-        config=config,
-    )
-
-    config.sep_token_id = tokenizer.convert_tokens_to_ids(tokenizer.sep_token)
-    config.output_dropout_rate = task_args.output_dropout_rate
-    config.decoding_times_with_dropout = task_args.decoding_times_with_dropout
-    model = ModelForDownstreamTasks.from_config(
-        config,
-        model_args.model_name_or_path,
-        task_heads=task_heads,
-        cache_dir=model_args.cache_dir,
-    )
-    model.set_task_head(next(iter(task_heads)))
-
-    # load data
-    logger.info('Loading dataset')
-    if data_args.train_file is not None or data_args.eval_file is not None:
-        data_files = {}
-
-        if data_args.train_file is not None: 
-            data_files['train'] = glob.glob(data_args.train_file)
-        if data_args.eval_file is not None: 
-            data_files['validation'] = glob.glob(data_args.eval_file)
-
-        raw_datasets = datasets.load_dataset(data_args.data_file_format, 
-            data_files=data_files,
-            cache_dir=model_args.cache_dir)
+    if task_args.modality=="table":
```

Comment:
Is this sustainable? How many modalities do we expect to have?

Reply:
In future, we might expect image as another modality.